spirekey/src/middleware.ts

```diff
@@ -0,0 +1,77 @@
+import { hash } from '@kadena/cryptography-utils';
+import type { NextRequest } from 'next/server';
+
+const createClientId = (request: NextRequest): string => {
+  const str = `${request.headers.get('user-agent')}${request.headers.get('sec-ch-ua-platform')}`;
+  return hash(str);
+};
+
+// This function can be marked `async` if using `await` inside
+export async function middleware(request: NextRequest) {
+  if (!process.env.GA_ACCOUNT || !process.env.GA_ACCOUNT_API) {
+    throw new Error('google account and api secret are missing');
+  }
+
+  const clientId = createClientId(request);
+  const data = JSON.stringify({
+    client_id: clientId, // A unique client identifier
+    user_id: clientId, // A unique client identifier
+    events: [
+      {
+        name: 'page_view',
+        params: {
+          page_title: 'Spirekey',
+          page_location: request.url,
+          time_to_section: 123456,
+          send_page_view: true,
+        },
+      },
+    ],
+  });
+
+  const options = {
+    v: 2,
+    tid: `${process.env.GA_ACCOUNT}`,
+    gtm: 'a64e392f914fbb0fb2f065aad16be630',
+    _p: 1,
+    cid: clientId,
+    ul: 'en-gb',
+    sr: '1980x1080',
```

Comment:
we might be able to use `Viewport-Width` header to get some idea

Reply:
we dont have that in the request object